src/org/gk/slicing/SlicingEngine.java

```diff
@@ -228,15 +294,196 @@ public void slice() throws Exception {
+        // and eleven members. By changing the compartment of one of these members (e.g. zofenopril),
+        // ACEI pro-drugs should then have two compartments.
+        GKInstance entity = testDBA.fetchInstance(9619112L);
+        GKInstance member = testDBA.fetchInstance(9619097L);
+        GKInstance cytoplasm = testDBA.fetchInstance(459L);
+		originalCompartment = (GKInstance) member.getAttributeValue(ReactomeJavaConstants.compartment);
+        member.setAttributeValue(ReactomeJavaConstants.compartment, cytoplasm);
+
+        // Control.
+		attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+        assertEquals(1, attrValue.size());
+
+        // Populate the compartments.
+        populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+        attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+        assertEquals(2, attrValue.size());
+        
+        // Remove the added compartment.
+        member.setAttributeValue(ReactomeJavaConstants.compartment, originalCompartment);
+        populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+        attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+		// System.out.println(attrValue);
+        assertEquals(1, attrValue.size());
+
+		/*
+		 * Changing the compartments of the containing entity sets.
+		 */
+    	// Each of these EntitySet's (and all of their respective members) have just one compartment.
+    	List<Long> dbIds = Arrays.asList(9619112L,  // ACEI pro-drugs [extracellular region]
+										 182588L,   // INK4 [cytolsol]
+										 3215203L); // ING [nucleoplasm]
+
+    	// Compartment not included in the EntitySet's above.
+    	GKInstance plasmaMembrane = testDBA.fetchInstance(876L);
+
+		for (Long dbId : dbIds) {
+			entity = testDBA.fetchInstance(dbId);
+			originalCompartment = (GKInstance) entity.getAttributeValue(ReactomeJavaConstants.compartment);
+
+    		// Control.
+			attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+    		assertEquals(1, attrValue.size());
+
+    		// Fill attribute values.
+			populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+			entity.addAttributeValue(ReactomeJavaConstants.compartment, plasmaMembrane);
+			attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+			System.out.println(attrValue);
+    		assertEquals(2, attrValue.size());
+
+    		// Remove added attribute values.
+			entity.setAttributeValue(ReactomeJavaConstants.compartment, originalCompartment);
+			populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+			attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+    		assertEquals(1, attrValue.size());
+    	}
+    }
+    
+    /**
+     * 
+     * @param msg
+     * @param ps
+     */
+    static void logAndPrintln(String msg, PrintStream ps) {
```

Comment:
Please don't add a static method here.

Reply:
Fixed in [c1aeaf3](https://github.com/reactome/CuratorTool/pull/20/commits/c1aeaf38b382c1684797a68493f8059eab5e856b).